app/controllers/lab_tests_controller.rb

```diff
@@ -80,20 +88,55 @@ def destroy
 
   private
 
-  # Use callbacks to share common setup or constraints between actions.
+  def handle_success_response
+    flash[:notice] = t('.success')
+    redirect_to(@health_record)
+  end
+
+  def handle_error_response(message = nil)
+    flash.now[:alert] = message if message
+    load_error_dependencies
+    render :new, status: :unprocessable_entity
+  end
+
+  def determine_redirect_path
+    @health_record || @lab_test
+  end
+
+  def load_error_dependencies
+    return if @lab_test.biomarker_id.blank?
+
+    @selected_biomarker = Biomarker.find(@lab_test.biomarker_id)
+    @reference_ranges = @selected_biomarker.reference_ranges
+  end
+
+  def set_biomarkers
+    @biomarkers = Biomarker.all
+  end
+
   def set_lab_test
     @lab_test = LabTest.find(params[:id])
   end
 
-  def build_lab_test
-    @lab_test = current_user.lab_tests.build(lab_test_params)
```

Comment:
Why have you removed this method and add this line to the [action](https://github.com/Code-the-Dream-School/health-keeper/pull/23/files#diff-f7fcfdbad10e32bb265e9ba7415ebc17190f6dbd815cca2caeed74a00af4b6efR36)?

It was extracted from the method on purpose to support Single responsibility and decrease cognitive load.
Please consider moving this back.